### ISSUE TITLE
Precision and Recall is always 1, even for bad model

### ISSUE THREAD

**Issue by santorre** [ultralytics/ultralytics#17213]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have tried training with both yolov8 and yolov10. When I run yolov10 everything seems to work fine, except for that the model is overfitting after around 20 epochs. But when I train using yolov8 instead, the precision and recall in self.metrics is always exactly 1 in every epoch, even if the model trained is not good from the start.

I don't understand why this error occurs and why only for yolov8 and not yolov10? In which script is the precision and recall set? And how can I make sure that this problem does not occur, since it makes it impossible for me to properly validate my model during training

**Comment by UltralyticsAssistant** [comment]
👋 Hello @santorre, thank you for reaching out and contributing to Ultralytics 🚀! We understand that you're experiencing an issue with precision and recall metrics during training with YOLOv8.

If this is a 🐛 Bug Report, we kindly ask you to provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate the issue further.

In the meantime, please ensure your repository is up to date. You can upgrade to the latest `ultralytics` package with all dependencies in place using:

```bash
pip install -U ultralytics
```

For detailed documentation and examples, you might want to check out our [Docs](https://docs.ultralytics.com/). If you've got questions related to custom training, please include as much information as possible: dataset samples, training logs, and ensure you're adhering to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Connect with the Ultralytics community to share insights and learn from others. You might like the real-time chat on [Discord](https://ultralytics.com/discord) 🎧, engage in discussions on [Discourse](https://community.ultralytics.com), or contribute to threads on our [Subreddit](https://reddit.com/r/ultralytics).

An Ultralytics engineer will also look into your issue shortly, so you're in good hands. Thanks for your patience and enthusiasm in improving our tools! 😊

## Environments

YOLO can be run in several verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. Refer to [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. Refer to the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, assuring our various modes and tasks function correctly across multiple platforms.

**Comment by Y-T-G** [comment]
What's the code you're using for training?

How many images do you have?

**Comment by santorre** [comment]
@Y-T-G I have about 115000 images for training and 60000 for validation. The code I have problems with is the regular yolov8 (8n) code version 8.1.34

**Comment by Y-T-G** [comment]
That's not the official `ultralytics` 

It's the YOLOv10 fork. You should use the latest `ultralytics`

**Comment by Y-T-G** [comment]
Also that's too many images for validation. You should only use around 5k and move the rest to the training set.
